lua/lvim/plugin-loader.lua

```diff
@@ -106,14 +106,21 @@ function plugin_loader.load(configurations)
         end
       end
     end)
-    -- colorscheme must get called after plugins are loaded or it will break new installs.
-    vim.g.colors_name = lvim.colorscheme
-    vim.cmd("colorscheme " .. lvim.colorscheme)
   end, debug.traceback)
+
   if not status_ok then
     Log:warn "problems detected while loading plugins' configurations"
     Log:trace(debug.traceback())
   end
+
+  -- ref: https://github.com/neovim/neovim/issues/18201#issuecomment-1104754564
+  local colors = vim.api.nvim_get_runtime_file(("colors/%s.*"):format(lvim.colorscheme), false)
+  if #colors == 0 then
+    Log:warn(string.format("Could not find '%s' colorscheme", lvim.colorscheme))
+    return
+  end
+  vim.g.colors_name = lvim.colorscheme
+  vim.cmd("colorscheme " .. lvim.colorscheme)
```

Comment:
we should now move this into `theme.lua` since #3133 got merged

```diff
diff --git a/lua/lvim/core/theme.lua b/lua/lvim/core/theme.lua
index 394963a0..7a4c469b 100644
--- a/lua/lvim/core/theme.lua
+++ b/lua/lvim/core/theme.lua
@@ -1,5 +1,6 @@
 local M = {}
 
+local Log = require "lvim.core.log"
 M.config = function()
   lvim.builtin.theme = {
     name = "tokyonight",
@@ -85,17 +86,24 @@ end
 M.setup = function()
   -- avoid running in headless mode since it's harder to detect failures
   if #vim.api.nvim_list_uis() == 0 then
-    local Log = require "lvim.core.log"
     Log:debug "headless mode detected, skipping running setup for lualine"
     return
   end
 
   local status_ok, theme = pcall(require, "tokyonight")
-  if not status_ok then
+  if status_ok and theme then
+    theme.setup(lvim.builtin.theme.options)
+  end
+
+  -- ref: https://github.com/neovim/neovim/issues/18201#issuecomment-1104754564
+  local colors = vim.api.nvim_get_runtime_file(("colors/%s.*"):format(lvim.colorscheme), false)
+  if #colors == 0 then
+    Log:warn(string.format("Could not find '%s' colorscheme", lvim.colorscheme))
     return
   end
 
-  theme.setup(lvim.builtin.theme.options)
+  vim.g.colors_name = lvim.colorscheme
+  vim.cmd("colorscheme " .. lvim.colorscheme)
 
   require("lvim.core.lualine").setup()
```